renderer/Interaction.cpp

```diff
@@ -1130,6 +1187,15 @@ bool idInteraction::IsPotentiallyVisible( idScreenRect &shadowScissor ) {
 	return true;
 }
 
+void idInteraction::LinkPreparedSurfaces() {
+	for (auto it : surfsToLink) {
+		// actually link it in
+		it.surf->nextOnLight = *it.link;
+		*it.link = it.surf;
+	}
+	surfsToLink.Clear();
```

Comment:
`Clear` frees resources. Is it indended?
To be honest, allocating buffer for every interaction every frame sounds like a performance problem.

Perhaps the best solution would be to have storage for a few elements embedded into `idInteraction`, and switch to heap buffer only when this capacity is exceeded. Unfortunately, I don't see such container in idLib... that's the mindset of "we can always put hard cap on any array" =(

The simpler solution would be maintaining one large array instead of per-interaction arrays, but that would need synchronization. Although malloc calls are also working with single resource and thus doing synchronization, although with atomics only.

Reply:
No, not intentional :( I expected std::vector behaviour which just removes the elements from the list, not delete the underlying storage...

I guess I could opt for the manual linked list, since I can reuse the existing `nextOnLight` pointer. Not an elegant solution, either, but at least it's an established pattern. Let's see if it makes any difference in performance.

Reply:
Well, I exchanged it for linked lists, which makes the interface and code a tad uglier, but avoids the allocation issue. However, far as I can tell, it makes no measurable difference in-game. Let me know what you think.

Reply:
Well `idList::SetNum(0)` does not free memory. Linked lists are OK, I guess --- they will traversed only once, and won't contain much data.